Add ganalytics

pawelru opened the pull request.

dinakar29 commented on inst/pkgdown/_pkgdown.yml:
```diff
@@ -8,3 +8,5 @@ template:
       <img src="https://github.com/insightsengineering/hex-stickers/blob/main/PNG/nest.png?raw=true" alt="NEST" style="height: 30px;">
       <a href="https://pharmaverse.org/"><img src="https://github.com/insightsengineering/hex-stickers/blob/main/PNG/pharmaverse.png?raw=true" alt="pharmaverse" style="height: 30px;"></a>
       </div>
+  params:
+    ganalytics: "UA-125641273-1"
```
This only works for BS3. For BS5, you need something like this: https://github.com/insightsengineering/r.pkg.template/blob/main/_pkgdown.yml#L3-L15

dinakar29 replied:
Also a good idea to add consent mode: https://developers.google.com/tag-platform/devguides/consent for compliance purposes.

pawelru replied:
```suggestion
  includes:
    in_header: |
       <script>
         // Define dataLayer and the gtag function.
         window.dataLayer = window.dataLayer || [];
         function gtag(){dataLayer.push(arguments);}
         // Default ad_storage to 'denied' as a placeholder
         // Determine actual values based on your own requirements
         gtag('consent', 'default', {
           'ad_storage': 'denied'
         });
       </script>
       <!-- Global site tag (gtag.js) - Google Analytics -->
       <script async src="https://www.googletagmanager.com/gtag/js?id=UA-125641273-1"></script>
       <script>
         window.dataLayer = window.dataLayer || [];
         function gtag(){dataLayer.push(arguments);}
         gtag('js', new Date());
         gtag('config', 'UA-125641273-1');
       </script>
```

pawelru commented on inst/pkgdown/_pkgdown.yml:
```diff
@@ -8,3 +8,23 @@ template:
+         function gtag(){dataLayer.push(arguments);}
+         // Default ad_storage to 'denied' as a placeholder
+         // Determine actual values based on your own requirements
+         gtag('consent', 'default', {
+           'ad_storage': 'denied'
+         });
```
@dinakar29 
copy-pasted from the reference you provided
do you have any ideas for meaningful defaults?

pawelru commented on inst/pkgdown/_pkgdown.yml:
```diff
@@ -8,3 +8,23 @@ template:
       <img src="https://github.com/insightsengineering/hex-stickers/blob/main/PNG/nest.png?raw=true" alt="NEST" style="height: 30px;">
       <a href="https://pharmaverse.org/"><img src="https://github.com/insightsengineering/hex-stickers/blob/main/PNG/pharmaverse.png?raw=true" alt="pharmaverse" style="height: 30px;"></a>
       </div>
+  includes:
```
```suggestion
```

dinakar29 reviewed (approved):
LGTM

pawelru commented:
Thanks @dinakar29 I simplified your solution as I think that modal is "too heavy" to handle such thing. I replaced it with a notification-type of element. I tried to keep your JS logic. Please have a look.

I also removed symlink as I think it's not required. Please correct me if I am wrong.

I cannot test it in the locally generated docs - i.e. the modal / notification keeps appearing in all the pages regardless of my action (accept / cancel). Is that because of local / remote source of files?

dinakar29 commented:
Thanks for simplifying, @pawelru - but the core behavior also seems to be modified with your changes as the GA cookie doesn't seem to be set even though I click "Accept". UI updates are good, but I'll need to fix it so that the proper cookies are set on acceptance.

pawelru commented on inst/pkgdown/assets/consent.js:
```diff
@@ -0,0 +1,95 @@
+
+        function setChoice(e) {
+            e.preventDefault();
+            if ($(this).id == 'cookie_accept') {
```
```suggestion
            if (this.id == 'cookie_accept') {
```

dinakar29 commented:
Let's give this a try as a 1st iteration. We can always modify if something's wrong.
